src/oscar/apps/checkout/mixins.py

```diff
@@ -93,8 +95,37 @@ def add_payment_event(self, event_type_name, amount, reference=''):
     def generate_order_number(self, basket):
         """
         Return a new order number
+
+        CodeRed Comment: We are overwriting this function to check and make sure
+        there isn't an already existing order with the generated order number.
+        For an unknown reason, this has been shown to be possible, but unlikely.  
+
+        If there is an existing order with the order number, remake the basket
+        and make the order number from the new basket.
         """
-        return OrderNumberGenerator().order_number(basket)
+        order_number = OrderNumberGenerator().order_number(basket)
+        if Order._default_manager.filter(number=order_number).exists():
+            new_basket = Basket(
+                owner=basket.owner,
+                status=basket.status,
+                date_merged=basket.date_merged,
+                date_submitted=basket.date_submitted,
+                is_taxable=basket.is_taxable,
+                )
+            new_basket.strategy = basket.strategy
+            new_basket.save()
+
+            for line in basket.lines.all():
+                line.basket = new_basket
+                line.save()
+
+            new_basket.vouchers = basket.vouchers.all()
+
+            self.request.basket = new_basket
+            basket.delete()
```

Comment:
Question... if the `basket` already has an order number associated with it... do we actually want to delete it? (could be a valid order). Or would it make more sense to set it to some kind of closed/completed status?

Reply:
Orders don't have to have a basket associated with them.  So it probably doesn't matter either way.  It doesn't appear that Orders use baskets for anything after they're done processing.